astropy/io/fits/file.py

```diff
@@ -400,6 +404,36 @@ def _overwrite_existing(self, overwrite, fileobj, closed):
             else:
                 raise IOError("File {!r} already exists.".format(self.name))
 
+    def _test_for_compression(self, obj_or_name, magic, mode, ext=None):
```

Comment:
That's a reasonable change, although I'll admit that I'm not sure I see a technical reason why comparison with `None` in this case is a problem.